### PR TITLE
Workers can now be started exclusively from the daemon and other workers

### PR DESCRIPTION
**Pull request by annando**
The daemon can now run on a different server than the frontend. The inter process communication is done via a memory table.

### CODE REVIEW

File: src/Core/Worker.php [hunk 1]
```diff
@@ -1121,4 +1140,30 @@ public static function endProcess()
 	{
 		return Process::deleteByPid();
 	}
+
+	private static function checkIPC()
+	{
+		dba::e("CREATE TABLE IF NOT EXISTS `worker-ipc` (`key` integer, `jobs` boolean) ENGINE = MEMORY;");
```
Comment (MrPetovan) [inline src/Core/Worker.php]: Shouldn't this be in DBStructure.php?

Comment (annando) [reply src/Core/Worker.php]: We weren't able to create memory tables. Now we are.

**Comment by annando** [comment]
@nupplaphil This PR here is **exactly** for that purpose. Means: You can now set up two different servers with the one server running the web process and the other one the daemon. The daemon then forks all needed workers.

To do this, you only have to deactivate the cron call and start the daemon. Everything else is done automatically.

**Comment by nupplaphil** [comment]
@annando I don't want to use the daemon because the process has to run in foreground.

Have a look at:
https://github.com/friendica/docker/blob/master/docker-cron.sh

Therefore I have to use `bin/worker.php` and not the daemon. Because it doesn't seem to be usable to use a daemon for a foreground process :-)

**Comment by annando** [comment]
The daemon is running in the background. It is started in the shell.

**Comment by annando** [comment]
Is there something I should do, before this PR can be accepted?

**Comment by annando** [comment]
But that was only a comment about the daemon itself, not about this special chance, wasn't it?

**Comment by MrPetovan** [comment]
Special chance? And I'm not sure about it, I'd rather him to acknowledge your comment and confirm it's good on his end.

**Comment by annando** [comment]
Okay. This PR isn't blocking me, so it's okay.

**Comment by nupplaphil** [comment]
Oh, I'm sorry I thought about making a follow up PR for the worker.php itself to not hijacking this PR. @annando is right that this daemon is a background process and this is fine for this PR.

So for me, the new inter process communication is great for running seperate instances for the same node.

But I do want to go further to use them for the `bin/worker.php` too.

So my only requirement for this PR would be that this functions are compatible for a further, similar change for `bin/worker.php` too :-)

**Comment by annando** [comment]
Question is: In which way should the workers speak with each other? How do they benefit from this?

**Comment by nupplaphil** [comment]
I thought about:

1) communicating between the APP and a cron process (in different instances)

2) communicating between different cron instances (for vertical scaling)

3) communicating between different APP instances (loadbalancing)

**Comment by nupplaphil** [comment]
I will make a diagram/concept about what I want to do and share it with the friendica develop forum :-)

Maybe I can describe it better what I mean :-)

**Comment by annando** [comment]
Okay. I will have a look at it. Currently the system already does some load depending scaling. And it already works with semaphore locks. The whole concept is based on the idea of a headless coordination.

**Pull request merged**